spec/ParseRole.spec.js

```diff
@@ -211,26 +155,35 @@ describe('Parse Role testing', () => {
   });
 
   it("Should properly resolve roles", (done) => {
-    const admin = new Parse.Role("Admin", new Parse.ACL());
-    const moderator = new Parse.Role("Moderator", new Parse.ACL());
-    const superModerator = new Parse.Role("SuperModerator", new Parse.ACL());
-    const contentManager = new Parse.Role('ContentManager', new Parse.ACL());
-    const superContentManager = new Parse.Role('SuperContentManager', new Parse.ACL());
+    const admin = new Parse.Role("Admin", createSelfAcl("Admin"));
+    const moderator = new Parse.Role("Moderator", createSelfAcl("Moderator"));
+    const superModerator = new Parse.Role("SuperModerator",createSelfAcl("SuperModerator"));
+    const contentManager = new Parse.Role('ContentManager', createSelfAcl("ContentManager"));
+    const superContentManager = new Parse.Role('SuperContentManager', createSelfAcl("SuperContentManager"));
     Parse.Object.saveAll([admin, moderator, contentManager, superModerator, superContentManager], {useMasterKey: true}).then(() => {
       contentManager.getRoles().add([moderator, superContentManager]);
       moderator.getRoles().add([admin, superModerator]);
       superContentManager.getRoles().add(superModerator);
       return Parse.Object.saveAll([admin, moderator, contentManager, superModerator, superContentManager], {useMasterKey: true});
     }).then(() => {
-      const auth = new Auth({ config: Config.get("test"), isMaster: true });
+      const auth = new Auth({ config: Config.get("test"), isMaster: false });
+      // For each role, create a user that
       // For each role, fetch their sibling, what they inherit
       // return with result and roleId for later comparison
       const promises = [admin, moderator, contentManager, superModerator].map((role) => {
-        return auth._getAllRolesNamesForRoleIds([role.id]).then((result) => {
+        const authRoles = auth.getAuthRoles()
```

Comment:
Ok so this test fails because the roles are not readable by any one so not applied